nahusenayElias/Team_4
# DRD-90: Mautic form to Drupal and from Drupal to frontend

## Related ticket
[DRD-90](https://edu-team4-react24k.atlassian.net/jira/software/projects/DRD/boards/1?selectedIssue=DRD-90)

## In this ticket: 
- Made changes to the Mautic paragraph module so that it uses the localhost of the Mautic instead of appserver.mauticapp.internal to display the forms in Drupal
- Added a Mautic contact form block
- Fetching the form ID from Drupal and rendering the form to the frontend with embedded Javascript script tag instead of the self-hosted version.

## Testing instructions:
- checkout branch
- lando db-import DRD-90.gz
- lando drush cim
- lando drush cr
- Go to Drupal > Configuration > Web services > Mautic integration > Add your Mautic localhost URL to the Mautic public url field
![image](https://github.com/user-attachments/assets/6b17fafa-e417-4568-91d7-a01473ce03e6)
- check that you can see the form in content below section of the page (the whole form is not displayed for some reason)
![image](https://github.com/user-attachments/assets/8d1ee4cf-222a-4858-a14e-f2e8540e0686)
- check that frontend is also displaying the form correctly and form submitting still works
![image](https://github.com/user-attachments/assets/ba2699a7-61a4-4dc7-bb3f-f0af7512795b)



[DRD-90]: https://edu-team4-react24k.atlassian.net/browse/DRD-90?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
